### PR TITLE
Deduplicate contributors configuration

### PR DESCRIPTION
**Pull request by max-ostapenko**
#2246

1. See new `config/contributors.json` and updated `config/{year}.json`.
A script used to merge contributors configuration:
```python
import json
from collections import defaultdict


def merge_contributors():
    years = ["2019", "2020", "2021", "2022"]
    merged_contributors = defaultdict(lambda: defaultdict(dict))

    for year in years:
        filename = f"src/config/{year}.json"
        with open(filename, "r+") as file:
            data = json.load(file)
            contributors = data.get("contributors", {})

            for key, value in contributors.items():
                if "teams" not in merged_contributors[key]:
                    merged_contributors[key]["teams"] = {}
                merged_contributors[key]["teams"][year] = value.pop("teams")
                merged_contributors[key].update(value)

    with open("src/config/contributors.json", "w") as outfile:
        json.dump(merged_contributors, outfile, indent=2, sort_keys=True)
        print("", file=outfile)
```

2. Adjusted the following scripts to be compatible with new configs:
- [x] tools/generate/shared.js
- [x] server/config.py

### CODE REVIEW

File: src/config/contributors.json [hunk 1]
```diff
@@ -0,0 +1,4295 @@
...
+    "twitter": "mel_melificent"
+  },
+  "meyerweb": {
+    "avatar_url": "939727",
+    "github": "meyerweb",
+    "name": "Eric A. Meyer",
+    "teams": {
+      "2019": [
+        "reviewers",
+        "editors"
```
Comment (tunetheweb) [inline src/config/contributors.json:2855]: This looks to be a mistake - see #3400

```suggestion
        "reviewers"
```

**Review by tunetheweb — CHANGES_REQUESTED** [review]
This looks very good - thanks for doing it @max-ostapenko !

A couple of small changes, but overall I'm very impressed how nice you've made this.

File: src/config/merge_contributors.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import json
```
[the remark this answers is not in the document]
Comment (max-ostapenko) [reply src/config/merge_contributors.py]: Yeap, moved it to the conversation.

File: src/config/2019.json [hunk 3]
```diff
@@ -2,8 +2,25 @@
...
+        "ru",
+        "tr",
+        "uk",
+        "zh-CN",
+        "zh-TW"
+      ],
+      "ebook_languages": [
+        "en",
+        "ja"
+      ]
```
Comment (tunetheweb) [inline src/config/2019.json:23]: Lets just collapse again, as pretty short:

```suggestion
      "supported_languages": ["en","es","fr","hi","it","ja","nl","pt","ru","tr","uk","zh-CN","zh-TW"],
      "ebook_languages": ["en","ja"]
```

File: src/config/2020.json [hunk 4]
```diff
@@ -2,8 +2,25 @@
...
+        "ru",
+        "tr",
+        "uk",
+        "zh-CN",
+        "zh-TW"
+      ],
+      "ebook_languages": [
+        "en",
+        "ja"
+      ]
```
Comment (tunetheweb) [inline src/config/2020.json:23]:
```suggestion
      "supported_languages": ["en","es","fr","hi","it","ja","nl","pt","ru","tr","uk","zh-CN","zh-TW"],
      "ebook_languages": ["en","ja"]
```

File: src/config/2021.json [hunk 5]
```diff
@@ -2,8 +2,25 @@
...
+        "ru",
+        "tr",
+        "uk",
+        "zh-CN",
+        "zh-TW"
+      ],
+      "ebook_languages": [
+        "en",
+        "ja"
+      ]
```
Comment (tunetheweb) [inline src/config/2021.json:23]:
```suggestion
      "supported_languages": ["en","es","fr","hi","it","ja","nl","pt","ru","tr","uk","zh-CN","zh-TW"],
      "ebook_languages": ["en","ja"]
```

File: src/server/config.py [hunk 6]
```diff
@@ -95,6 +101,12 @@ def update_config():
                 SUPPORTED_LANGUAGES.update({year: get_languages(json_config)})
                 SUPPORTED_CHAPTERS.update({year: set(get_chapters(json_config))})
 
+                json_config["contributors"] = {}
+                for contributor_id, contributor in contributors.items():
+                    if (year in contributor["teams"]):
+                        json_config["contributors"][contributor_id] = {**contributor}
+                        json_config["contributors"][contributor_id]["teams"] = contributor["teams"][year]
```
[the remark this answers is not in the document]
Comment (max-ostapenko) [reply src/server/config.py:110]: I think we can merge this one and adjust the scripts later when required.

File: src/config/2022.json [hunk 7]
```diff
@@ -2,8 +2,25 @@
...
+        "ru",
+        "tr",
+        "uk",
+        "zh-CN",
+        "zh-TW"
+      ],
+      "ebook_languages": [
+        "en",
+        "ja"
+      ]
```
Comment (tunetheweb) [inline src/config/2022.json:23]:
```suggestion
      "supported_languages": ["en","es","fr","hi","it","ja","nl","pt","ru","tr","uk","zh-CN","zh-TW"],
      "ebook_languages": ["en"]
```

File: src/config/2022.json [hunk 8]
```diff
@@ -39,34 +56,21 @@
         {
           "part": "I",
           "chapter_number": "5",
-          "title": "Fonts",
-          "slug": "fonts"
-        },
-        {
-          "part": "I",
-          "chapter_number": "6",
```
Comment (tunetheweb) [inline src/config/2022.json:47]: What's going on here? Looks like rest of this file messed up (copy and paste error?)

**Review by tunetheweb — CHANGES_REQUESTED** [review]

File: src/config/2019.json (hunk 3 above)
Comment (max-ostapenko) [reply src/config/2019.json:23]: wow, it's my "format on paste" IDE setting... Need to switch it off.

Comment (tunetheweb) [reply src/config/2019.json:23]: Your IDE is probably correct in most cases 😁

But in this case I think overkill.

**Review by tunetheweb — APPROVED** [review]
Looks great - thanks @max-ostapenko !!

**Pull request merged**